tests/phpunit/integration/Modules/Analytics_4Test.php

```diff
@@ -1666,6 +1667,262 @@ public function test_get_conversion_events( $access_token ) {
+				(array) Analytics::READONLY_SCOPE
+			)
+		);
+
+		FakeHttp::fake_google_http_handler(
+			$this->analytics->get_client(),
+			$this->create_fake_http_handler( $property_id )
+		);
+		$this->analytics->register();
+
+		// Fetch enhanced measurement settings.
+		$data = $this->analytics->get_data(
+			'enhanced-measurement-settings',
+			array(
+				'propertyID'      => $property_id,
+				'webDataStreamID' => $web_data_stream_id,
+			)
+		);
+
+		$this->assertNotWPError( $data );
+
+		$data_array = (array) $data;
+
+		// Assert that the keys exist.
+		$keys = array(
+			'fileDownloadsEnabled',
+			'name',
+			'outboundClicksEnabled',
+			'pageChangesEnabled',
+			'scrollsEnabled',
+			'searchQueryParameter',
+			'siteSearchEnabled',
+			'streamEnabled',
+			'uriQueryParameter',
+			'videoEngagementEnabled',
+		);
+
+		foreach ( $keys as $key ) {
+			$this->assertArrayHasKey( $key, $data_array );
+		}
+
+		// Verify the request URL and params were correctly generated.
+		$this->assertCount( 1, $this->request_handler_calls );
+
+		$request_url = $this->request_handler_calls[0]['url'];
+
+		$this->assertEquals( 'analyticsadmin.googleapis.com', $request_url['host'] );
+		$this->assertEquals( "/v1alpha/properties/$property_id/dataStreams/$web_data_stream_id/enhancedMeasurementSettings", $request_url['path'] );
+	}
+
+	public function test_set_enhanced_mesurement_settings__required_params() {
+		$property_id        = '123456789';
+		$web_data_stream_id = '654321';
+
+		FakeHttp::fake_google_http_handler(
+			$this->analytics->get_client(),
+			$this->create_fake_http_handler( $property_id )
+		);
+		$this->analytics->register();
+
+		// Call set_data without EDIT_SCOPE.
+		$data = $this->analytics->set_data(
+			'enhanced-measurement-settings',
+			array(
+				'propertyID'                  => $property_id,
+				'webDataStreamID'             => $web_data_stream_id,
+				'enhancedMeasurementSettings' => array(
+					'streamEnabled' => true,
+				),
+			)
+		);
+
+		// Verify if the EDIT_SCOPE is required.
+		$this->assertWPErrorWithMessage( 'You’ll need to grant Site Kit permission to update enhanced measurement settings for this Analytics 4 web data stream on your behalf.', $data );
+		$this->assertEquals( 'missing_required_scopes', $data->get_error_code() );
+		$this->assertEquals(
+			array(
+				'scopes' => array(
+					'https://www.googleapis.com/auth/analytics.edit',
+				),
+				'status' => 403,
+			),
+			$data->get_error_data( 'missing_required_scopes' )
+		);
+
+		// Grant EDIT_SCOPE so request doesn't fail.
+		$this->authentication->get_oauth_client()->set_granted_scopes(
+			array_merge(
+				$this->authentication->get_oauth_client()->get_required_scopes(),
+				(array) Analytics::EDIT_SCOPE
+			)
+		);
+
+		// Call set_data with no parameters.
+		$data = $this->analytics->set_data(
+			'enhanced-measurement-settings',
+			array()
+		);
+
+		// Verify if the propertyID is required.
```

Comment:
Same minor grammatical point applies:

```suggestion
		// Verify that the propertyID is required.
```